workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/DialogRunner.kt

```diff
@@ -0,0 +1,150 @@
+package com.squareup.workflow1.ui.container
+
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import android.os.Parcel
+import android.os.Parcelable
+import android.os.Parcelable.Creator
+import androidx.core.view.doOnAttach
+import androidx.core.view.doOnDetach
+import androidx.lifecycle.Lifecycle.Event.ON_DESTROY
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.OnLifecycleEvent
+import com.squareup.workflow1.ui.Compatible
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.androidx.WorkflowLifecycleOwner
+import com.squareup.workflow1.ui.compatible
+
+@WorkflowUiExperimentalApi
+internal class DialogRunner<T : Overlay>(
```

Comment:
I would change this to `DialogHolder`. I realize we use `Runner` elsewhere but I also just realized that while I got the 'Runner' metaphor the 'ViewHolder' analogy (as in the RecyclerView sense) is much more powerful to me as an Android person with RecyclerView experience.

Reply:
Great call. When we use `Runner` elsewhere it really means something else entirely.